core/server/api_container/server/startosis_engine/instructions_plan/resolver/instructions_plan_mask.go

```diff
@@ -0,0 +1,43 @@
+package resolver
+
+import "github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/instructions_plan"
+
+type InstructionsPlanMask struct {
+	readIdx               int
+	scheduledInstructions []*instructions_plan.ScheduledInstruction
+	isValid               bool
+}
+
+func NewInstructionsPlanMask(size int) *InstructionsPlanMask {
+	return &InstructionsPlanMask{
+		readIdx:               0,
+		scheduledInstructions: make([]*instructions_plan.ScheduledInstruction, size),
+		isValid:               true, // the mask is considered valid until it's proven to be invalid
```

Comment:
@gbouv What made chose to assume it's valid? I guess my only concern is that if no validation is performed it would be a false positive?

Reply:
That's part of the heuristic I explain in the Interpreter code. We tentatively generate plans we think could be valid, then run an interpretation round with this plan to validate it's indeed valid. If it's proven to be invalid, then we continue with another tentative, and so on and so on until we run out of plan, in which case nothing can be cached and we just default to the behaviour we currently have (i.e. run starlark instructions on top of whatever exist in the current enclave).
This is why we initialise isValid to true.